lib/geocoder/lookups/base.rb

```diff
@@ -83,8 +83,13 @@ def query_url(query)
       # The working Cache object.
       #
       def cache
-        if @cache.nil? and store = configuration.cache
-          @cache = Cache.new(store, configuration.cache_prefix)
+        if @cache.nil? && (store = configuration.cache)
+          cache_options = configuration.cache_options
+          if configuration.cache_prefix.present?
```

Comment:
This functionality is good, but the `present?` method is provided by Rails, so we can't use it here.